lib/MetaCPAN/Web/Role/Elasticsearch/Adapter.pm

```diff
@@ -0,0 +1,50 @@
+package MetaCPAN::Web::Role::Elasticsearch::Adapter;
+
+use Moose::Role;
+
+=head1 METHODS
+
+=head2 extract_first_element
+
+Elasticsearch 1.x changed the data structure returned when fields are used.
+For example before one could get a ArrayRef[HashRef[Str]] where now
+that will come in the form of ArrayRef[HashRef[ArrayRef[Str]]]
+
+This function will select the first element given an array of hash of array.
+So this:
+
+    [
+      {
+        distribution => ['WhizzBang'],
+        author       => ['EVE'],
+      },
+       ...
+   ]
+
+becomes:
+
+    [
+      {
+        distribution => 'WhizzBang',
+        author       => 'EVE',
+      },
+      ...
+    ]
+
+=cut
+
+sub extract_first_element {
+    my ( $self, $array ) = @_;
+    foreach my $hash ( @{$array} ) {
+        foreach my $field ( %{$hash} ) {
+            my $value = $hash->{$field};
+
+            # We only extract the first element when have an ArrayRef
+            next if not( ref($value) and ( ref($value) eq 'ARRAY' ) );
```

Comment:
> So, if an adapter is safe for us to use, I'd be more than happy to merge
it into master and delete it after the upgrade.

Precisely why I suggested forward compatible rather than backward
compatible.  If we upgrade scalars to arrays (and fix the other code to
expect arrays) the code will be more future proof and the adapter can be
removed after the upgrade. If we instead downgrade arrays to scalars then
the adapter lives forever.